agent/src/api_client.py

```
@@ -1660,6 +1664,105 @@ async def _prepare_result_upload(
+                        f"Download failed with status {response.status_code}",
+                        status_code=response.status_code,
+                    )
+
+                hasher = hashlib.sha256()
+                with open(dest_path, "wb") as f:
+                    for chunk in response.iter_bytes(chunk_size=65536):
+                        f.write(chunk)
+                        hasher.update(chunk)
+
+        except httpx.ConnectError as e:
+            raise ConnectionError(f"Failed to connect to server: {e}")
```

Ruff (0.15.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)